components/ModEditSubmissionTopbar.vue

```diff
@@ -73,7 +75,41 @@ const acceptSubmission = () => {
     console.log('You are accepted')
 }
 
-const rejectSubmission = () => {
-    console.log('You are rejected')
+const rejectSubmission = async () => {
+    const facilityInputVariables = {
+        updateSubmissionId: selectedSubmissionId.value,
+        input: {
+            isRejected: true
+        }
+    }
+
+    const maxAttempts = 3
+    let mutationAttempts = 0
+    let mutationSuccess = false
+
+    function delay(milliseconds: number) {
+        return new Promise(resolve => setTimeout(resolve, milliseconds))
+    }
+
+    while (mutationAttempts < maxAttempts && !mutationSuccess) {
```

Comment:
love the retries! Since retries are a common pattern, usually most query libraries already have them built in. Let's see if our gql client has them.
We should also move the retry logic to the gqlclient so every request has the retry logic and can overridden through optional parameters. Otherwise, we're at the goodwill of other devs to write the same retry logic you have here

Reply:
Now we will have our own built in